.github/workflows/_pre-commit-update.yml

```diff
@@ -0,0 +1,106 @@
+name: Pre-Commit update
+
+permissions:
+  contents: read
+
+on:
+  workflow_call:
+    inputs:
+      update_branch:
+        description: Name of a branch that will be used to introduce changes
+        type: string
+        default: pre-commit-dependencies-update
+    outputs:
+      pr_number:
+        description: A PR number if one is created or updated
+        value: ${{ jobs.update.outputs.pr_number }}
+      pr_operation:
+        description: 'An operation taken on a PR, either: created, updated, closed, or null string in case no operation was made'
+        value: ${{ jobs.update.outputs.pr_operation }}
+
+jobs:
+  update:
+    name: "Update Pre-Commit dependencies"
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    outputs:
+      pr_number: ${{ steps.pr.outputs.pull-request-number }}
+      pr_operation: ${{ steps.pr.outputs.pull-request-operation }}
+    steps:
+      - name: checkout repository
+        uses: actions/checkout@v3
+
+      - name: install Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
+
+      - name: install Python requirements
+        shell: bash
+        run: python -m pip install pyyaml>=6 requests>=2
+
+      - name: run the update script
+        id: update
+        shell: python
+        run: | 
+          import os
+          import yaml
+          import requests
+          from urllib.parse import urlparse
+          
+
+          pre_commit_config_path = "./.pre-commit-config.yaml"
+
+          with open(pre_commit_config_path, "r") as pre_commit:
+              try:
+                  pre_commit_config = yaml.safe_load(pre_commit)
+              except yaml.YAMLError as exc:
+                  print(exc)
+                  exit(1)
+          
+          updates = dict()
+
+          for repo in pre_commit_config['repos']:
+              repo_url = repo['repo'].replace('.git','') if '.git' in repo['repo'] else repo['repo']
+              release = repo['rev']
+              headers = {'Accept': 'application/json'}
+              resp = requests.get(f"{repo_url}/refs?type=tag", headers=headers)
+              if resp.status_code == requests.codes.ok:
```

Comment:
Is it a good idea to run the check against `requests.codes.ok`? This might include status codes that are other than "200" - maybe we can just check the 200 status code directly - because if we get a 302 REDIRECT for example - the current code won't treat that situation.

Reply:
we shouldn't get redirects here - we request the main repo URL, if it's not there something is NOK